server/services/utils/helpers.js

```diff
@@ -26,3 +26,38 @@ export const isIndexNotFoundError = (err) => {
       'Configured indices are not found: [.opendistro-alerting-config]'
   );
 };
+
+export function getClientBasedOnDataSource(
+  context,
+  dataSourceEnabled,
+  dataSourceId,
+  request,
+  client
+) {
+  if (dataSourceEnabled && dataSourceId && dataSourceId.trim().length !== 0) {
+    // Client for remote cluster
+    return context.dataSource.opensearch.legacy.getClient(dataSourceId).callAPI;
+  } else {
+    // Fall back to default local cluster
+    return client.asScoped(request).callAsCurrentUser;
+  }
+}
+
+export function validateQuery(dataSourceEnabled, fields = {}) {
```

Comment:
I would rename this to something like `createValidateQuerySchema` since that is all we do here. Actual validation happens at run time